src/UI/SlotArea.h

```diff
@@ -273,6 +273,11 @@ class cSlotAreaCrafting:
 	// Distributing items into this area is completely disabled
 	virtual void DistributeStack(cItem & a_ItemStack, cPlayer & a_Player, bool a_ShouldApply, bool a_KeepEmptySlots, bool a_BackFill) override;
 
+	/** Clear the crafting grid */
+	void ClearCraftingGrid(cPlayer & a_Player);
+
+	/** Loads the given Recipe into the crafting grid */
+	void LoadRecipe(cPlayer & a_Player, UInt32 a_RecipeId);
```

Comment:
How about making these available in the base class, so that the plugins can use them for different windows / areas, too? It would get rid of the nasty `static_cast<cSlotAreaCrafting *>` in `cInventoryWindow`.

Reply:
Is this a rhetorical question?

The method is specific for the inventory and crafting slots, it does not make sense in e.g. the hotbar area. I don't like the duplicate code and the casting, but having in for all slot areas feels dirty, too.
I can change it, not my call.

Reply:
Well, would be cool, but `ClearCraftingGrid` relies on `m_GridSize` which is only available in `cSlotAreaCrafting`.